File: php/src/Model/LearnedWeightsPreview.php
<?php

declare(strict_types=1);

namespace Fivexer\SDK\Model;

use Fivexer\SDK\Internal\Json;

/**
 * What applying the learned weights would change.
 *
 * Response model: built from a decoded JSON array and read-only thereafter.
 */
final class LearnedWeightsPreview
{
    public function __construct(
        /** @var list<PreviewedWorkerWeights> */
        public readonly array $workers,
    ) {
    }

    /**
     * @param array<string, mixed> $data
     * @internal
     */
    public static function fromArray(array $data): self
    {
        return new self(
            workers: Json::parseEach($data, 'workers', [PreviewedWorkerWeights::class, 'fromArray']),
        );
    }
}

File: java/src/main/java/io/fivexer/sdk/model/LearnedWeightsPreview.java
package io.fivexer.sdk.model;

import java.util.List;

/**
 * What applying the learned weights would change.
 *
 * <p>Populated by Gson via field reflection and exposed read-only; callers receive instances
 * from the client and never construct them.
 */
public class LearnedWeightsPreview {
    private List<PreviewedWorkerWeights> workers;

    public List<PreviewedWorkerWeights> getWorkers() { return workers; }
}
